### PR TITLE
Cfg override doc update

### PR DESCRIPTION
**Pull request by tbarnes-us**

### CODE REVIEW

File: docs-source/content/userguide/managing-domains/configoverrides/_index.md [hunk 1]
```diff
@@ -122,7 +124,7 @@ The operator requires a different file name format for override templates than W
 | JDBC module     |  `jdbc-MODULENAME.xml`  |
 | Diagnostics module     |  `diagnostics-MODULENAME.xml`  |
 
-A `MODULENAME` must correspond to the MBean name of a system resource defined in your original `config.xml` file.
+A `MODULENAME` must correspond to the MBean name of a system resource defined in your original `config.xml` file. It's not possible to add a new module via overrides. If you need your overrides to setup a new module, then have your original configuration specify 'skeleton' modules that can be overridden.
```
Comment (rosemarymarano) [inline docs-source/content/userguide/managing-domains/configoverrides/_index.md]: via -> by using (Avoid. Use by, by using, or through)
setup -> set up (set up (v), setup (n) (adj))

File: docs-source/content/userguide/managing-domains/configoverrides/_index.md [hunk 2]
```diff
@@ -184,24 +186,28 @@ The secret macro `SECRETNAME` field must reference the name of a Kubernetes secr
...
+       * See the `public-address` stanza in  [Override template samples](#override-template-samples) for an example.
 * When overriding `config.xml`:
   * The XML namespace (`xmlns:` in the XML) must be exactly as specified in [Override template schemas](#override-template-schemas).
        * For example, use `d:` to reference `config.xml` beans and attributes, `f:` for `add` and `replace` `domain-fragment` verbs, and `s:` to reference the situational configuration schema.
   * Avoid specifying the domain name stanza, as this may cause some overrides to be ignored (for example, server-template scoped overrides).
 * When overriding modules:
   * It is a best practice to use XML namespace abbreviations `jms:`, `jdbc:`, and `wldf:` respectively for JMS, JDBC, and WLDF (diagnostics) module override files.
+  * A module must already exist in your original configuration if you want to override it; it's not possible to add a new module via overrides. If you need your overrides to setup a new module, then have your original configuration specify 'skeleton' modules that can be overridden.
```
Comment (rosemarymarano) [inline docs-source/content/userguide/managing-domains/configoverrides/_index.md]: via -> by using (please fix globally)
setup -> set up (please fix globally)

File: docs-source/content/userguide/managing-domains/configoverrides/_index.md [hunk 3]
```diff
@@ -184,24 +186,28 @@ The secret macro `SECRETNAME` field must reference the name of a Kubernetes secr
...
+  * See [Overriding a data source module](#overriding-a-data-source-module) for best practice advice. Note that similar advice applies generally too other module types.
```
Comment (rosemarymarano) [inline docs-source/content/userguide/managing-domains/configoverrides/_index.md]: too  -> to

File: docs-source/content/userguide/managing-domains/configoverrides/_index.md [hunk 4]
```diff
@@ -184,24 +186,28 @@ The secret macro `SECRETNAME` field must reference the name of a Kubernetes secr
...
+* Consider having your original configuration reference invalid usernames, passwords, and URLs:
```
Comment (rosemarymarano) [inline docs-source/content/userguide/managing-domains/configoverrides/_index.md]: usernames -> user names (Use two words, but username in syntax or code instructions is acceptable. Please fix globally.)

File: docs-source/content/userguide/managing-domains/configoverrides/_index.md [hunk 5]
```diff
@@ -240,6 +246,11 @@ The following `config.xml` override file demonstrates:
 
 The following `jdbc-testDS.xml` override template demonstrates setting the URL, user name, and password-encrypted fields of a JDBC module named `testDS` via `secret macros`.  The generated situational configuration that replaces the macros with secret values will be located in the `DOMAIN_HOME/optconfig/jdbc` directory.   The `password-encrypted` field will be populated with an encrypted value because it uses a secret macro with an `:encrypt` suffix.  The secret is named `dbsecret` and contains three keys: `url`, `username`, and `password`.
 
+Best practices for data source modules and their overrides:
+
+* A data source module must already exist in your original configuration if you want to override it; it's not possible to add a new module via overrides. If you need your overrides to setup a new module, then have your original configuration specify 'skeleton' modules that can be overridden. See the next two bullets for the typical contents of a skeleton data source module.
```
Comment (rosemarymarano) [inline docs-source/content/userguide/managing-domains/configoverrides/_index.md]: next two bullets -> next two bulleted items

File: docs-source/content/userguide/managing-domains/configoverrides/_index.md [hunk 6]
```diff
@@ -240,6 +246,11 @@ The following `config.xml` override file demonstrates:
...
+* Set your original (non-overridden) URL, username, and password to invalid values. This helps prevent accidentally starting a server without overrides, and then having the datasource successfully connect to a database that's wrong for the current environment. For example, if these attributes are set to reference a QA database in your original configuration, then a mistake configuring overrides in your production Kubernetes deployment could cause your production applications to use your QA database.
```
Comment (rosemarymarano) [inline docs-source/content/userguide/managing-domains/configoverrides/_index.md]: datasource -> data source

File: docs-source/content/userguide/managing-domains/configoverrides/_index.md [hunk 7]
```diff
@@ -240,6 +246,11 @@ The following `config.xml` override file demonstrates:
...
+* Set your original (non-overridden) `JDBCConnectionPoolParams` `MinCapacity` and `InitialCapacity` to `0`, and set your original `DriverName` to a reference an existing JDBC Driver. This ensures you can still successfully boot a server even when you have configured invalid URL/username/password values, your database isn't running, and/or you haven't been specified your overrides yet.
```
Comment (rosemarymarano) [inline docs-source/content/userguide/managing-domains/configoverrides/_index.md]: ensures you -> ensures that you
you haven't been specified -> you haven't specified

**Comment by tbarnes-us** [comment]
I've addressed Rosemary's edit requests.

**Pull request merged**